// @asgardeo/auth-spa/dist/src/utils/message-utils.js
/**
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export class MessageUtils {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    constructor() { }
    /**
     * JSON stringifies the passed object.
     *
     * @param {any} data The data object.
     *
     * @return {ResponseMessage<string>} JSON string.
     */
    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    static generateSuccessMessage(data) {
        return {
            blob: (data === null || data === void 0 ? void 0 : data.data) instanceof Blob ? data === null || data === void 0 ? void 0 : data.data : null,
            data: JSON.stringify(data !== null && data !== void 0 ? data : ""),
            success: true
        };
    }
    /**
     *
     * Explicitly constructs a serializable error shape instead of relying on
     * JSON.stringify(error) directly. This is necessary because axios 1.x defines
     * toJSON() on the AxiosError prototype which deliberately excludes `response`
     * (to avoid leaking sensitive data). Since toJSON() is on the prototype and not
     * the instance, `delete error.toJSON` is a no-op, and JSON.stringify ends up
     * calling the prototype's toJSON — stripping response.data (and server-side
     * error codes like BPM-60006) from the postMessage payload.
     *
     * @param {any} error The error object.
     *
     * @return {ResponseMessage<string>} JSON string.
     */
    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    static generateFailureMessage(error) {
        var _a, _b, _c;
        const serializable = error ? {
            code: error === null || error === void 0 ? void 0 : error.code,
            config: error === null || error === void 0 ? void 0 : error.config,
            isAxiosError: error === null || error === void 0 ? void 0 : error.isAxiosError,
            message: error === null || error === void 0 ? void 0 : error.message,
            name: error === null || error === void 0 ? void 0 : error.name,
            request: error === null || error === void 0 ? void 0 : error.request,
            response: (error === null || error === void 0 ? void 0 : error.response) ? {
                config: error.response.config,
                data: error.response.data,
                headers: error.response.headers,
                request: error.response.request,
                status: error.response.status,
                statusText: error.response.statusText
            } : undefined,
            status: (_c = (_a = error === null || error === void 0 ? void 0 : error.status) !== null && _a !== void 0 ? _a : (_b = error === null || error === void 0 ? void 0 : error.response) === null || _b === void 0 ? void 0 : _b.status) !== null && _c !== void 0 ? _c : null
        } : "";
        return {
            error: JSON.stringify(serializable),
            success: false
        };
    }
}
